platform/osgi/platform-osgi-configadmin/src/test/java/org/codice/felix/cm/file/EncryptingWrappedPersistenceManagerTest.java

```diff
@@ -0,0 +1,79 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.felix.cm.file;
+
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+import static org.mockito.Mockito.when;
+import static org.osgi.framework.Constants.SERVICE_PID;
+import static org.osgi.service.cm.ConfigurationAdmin.SERVICE_FACTORYPID;
+
+import java.util.Dictionary;
+import java.util.Hashtable;
+
+import org.apache.felix.cm.PersistenceManager;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import ddf.security.encryption.EncryptionService;
+
+@RunWith(MockitoJUnitRunner.class)
+public class EncryptingWrappedPersistenceManagerTest {
+    private static final String PID = "my_pid";
+
+    private static final String VALUE_TO_ENCRYPT = "some_string";
+
+    private static final Dictionary<String, Object> TEST_CASES = new Hashtable<>();
+
+    static {
```

Comment:
❗️ We might want to add string properties that reflects encrypted and non-encrypted passwords.
I am also not comfortable to mock the encryption service as we are not really testing what would happen when it receives a special encrypted passwords (e.g. "enc(xxxx)") and decrypted passwords (e.g. "dec(xxx)").

Reply:
The encryption service does not automatically handle the wrapping for you. It simply encrypts what is provided to it in an attempt to least astonish us. There are explicitly named methods in the service that can help with this, but they must be invoked only if needed. 

As far as my encryption is concerned, I don't care what format I'm given. Even if I encrypt something that's already encrypted, I'll return it in its original encrypted form. The services that use those values know how to decrypt it back to plaintext without my help.

Reply:
Fair enough. We can still add 2 additional entries to your map for your round trip test case to verify that